### PR TITLE
Ruby group capstone - Catalog of my things

### PR DESCRIPTION
**Pull request by ridwanediallo**
## Project requirements
1. Each class from the [UML class diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png) should be created in a separate .rb file.
2. All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes should be implemented.
3. All properties visible in the diagram should be defined and set up in the constructor method. Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created.
Implement  methods visible in the diagram 
4. Add unit tests for all implemented methods.
5. Create a main.rb file that will serve as your console app entry-point.
6. All data should be preserved by saving collections in .json files.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    publish_date = take_input('Publish_date')
+    on_spotify = take_input('On_spotify?')
+    insert_music(name, publish_date, on_spotify)
+  end
+
+  def list_music
+    @music_Album.each do |music|
+      puts "[#{music.class}] Name: #{music.name} Publish_date: #{music.publish_date} On_spotify: #{music.on_spotify}"
+    end
+  end
```
Comment (ahmetbozaci) [inline app.rb:85]: - [ ] I am getting errors when try sees music albums. Kindly fix it.

image

File: README.md [hunk 2]
```diff
@@ -1,56 +1,40 @@
...
 
 👤 **Author2**
 
-- GitHub: [@githubhandle](https://github.com/githubhandle)
-- Twitter: [@twitterhandle](https://twitter.com/twitterhandle)
-- LinkedIn: [LinkedIn](https://linkedin.com/in/linkedinhandle)
+| 👤 Name             | Github                               | Twitter                                 | LinkedIn                                                  |
+| ------------------- | ------------------------------------ | --------------------------------------- | --------------------------------------------------------- |
+| abdelrahman mohamed | [@Abdo](https://github.com/Abdo9826) | [@Abdo](https://twitter.com/abodyalex1) | [@Abdo](https://www.linkedin.com/in/abdelruhman-mihamed/) |
+
```
Comment (ahmetbozaci) [inline README.md:32]: - [ ] To make a professional readme file please fix the broken link in the license section. 
_I can't choose the license line so ı choose an empty line. Kindly ignore it._

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# ♻️♻️♻️ STATUS: Changes Required  ♻️♻️♻️


 @ridwanediallo @Abdo9826 @shahnawaza75 ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Highlights 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
- You added tests and all tests passed.✔️
- Data is saved after quitting the app.✔️
___


### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._
- [ ] Your presentation video is good. You showed a demo of the project. But you didn't mention or show some codes that you are proud of it. Kindly add it. 
- [ ]  Also, the video length should be maxed at 5 minutes.
___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Review by andersonlebon — APPROVED** [review]
## Hi @ridwanediallo 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Optional suggestions
- [ ] Kindly Check the comments under the review._
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**